stubs/oauthlib/oauthlib/oauth2/rfc6749/request_validator.pyi

```diff
@@ -24,9 +37,9 @@ class RequestValidator:
     def invalidate_authorization_code(self, client_id: str, code: str, request: Request, *args, **kwargs) -> None: ...
     def revoke_token(self, token: str, token_type_hint: str, request: Request, *args, **kwargs) -> None: ...
     def rotate_refresh_token(self, request: Request) -> bool: ...
-    def save_authorization_code(self, client_id: str, code: str, request: Request, *args, **kwargs) -> None: ...
-    def save_token(self, token: dict[str, int | str], request: Request, *args, **kwargs) -> None: ...
-    def save_bearer_token(self, token: dict[str, int | str], request: Request, *args, **kwargs) -> str: ...
+    def save_authorization_code(self, client_id: str, code: _AuthorizationCode, request: Request, *args, **kwargs) -> None: ...
+    def save_token(self, token: _BearerToken | dict[str, Any], request: Request, *args, **kwargs) -> object: ...
```

Comment:
Why do you have the `| dict[str, Any]` here but not in the next method?

I'm of two minds about this. On the one hand, invariance means that it may be difficult for users to type their code correctly if we require a specific TypedDict with a non-exported type. On the other hand, giving `dict[str, Any]` as an option means that essentially any dict is accepted, so we lose any type safety from the TypedDict. On balance, I'd prefer to stick with just the TypedDict for now.

Reply:
Yeah, I am not totally happy about this either, I think the idea is that `save_token` accepts any token type and does the disambiguation and dispatch to the other `save_X_token` methods. Of course there's only the `Bearer` token right now in the RFC, so the function does not do any of that and just immediately calls `save_bearer_token`, but some implementations may have additional token types, so if they want to subclass this for that specific implementation they will have to ignore override errors.

That being said, `dict[str, Any]` is probably not the right way to go about this either, maybe it should just be `Mapping[str, object]` and then implementations can decide what to narrow it to, if anything. Or we just stick with the most common use-case and recognize that `save_token` doesn't have any safeguards, and eat the potential `override` errors.

Reply:
I went with `Mapping[str, object]`. I think this makes the most sense, since `save_token` should generally not be called manually unless you are implementing a non-standard OAuth workflow and in that case you may need to override `save_token` anyways, usually you will only override `save_bearer_token`.

I think what's being passed in is usually a `oauthlib.oauth2.rfc6749.tokens.Token` (although not always, sometimes it's just a plain `dict`) which is a subclass of `dict` and generated using `oauthlib.oauth2.rfc6749.tokens.BearerToken`. The problem is, that `Token` is generic so it can have any specific keys, but `BearerToken` does generate a dictionary with specific keys, so without intersections we can't really type this properly, so typing `save_bearer_token` strict and `save_token` lax feels like the best compromise for now.

Reply:
Thanks, we'll go with this for now and we can revisit it if users complain.